datajunction-ui/src/app/services/DJService.js

```diff
@@ -613,6 +613,51 @@ export const DataJunctionAPI = {
     );
     return { status: response.status, json: await response.json() };
   },
+
+  addComplexDimensionLink: async function (
+    nodeName,
+    dimensionNode,
+    joinType = 'left',
```

Comment:
Perhaps these defaults should just be left null, so that they can default to the backend defaults rather than relying on defaults embedded in the frontend? Of course, if we decide that we need different defaults when this is being accessed via the UI vs via an API call, we can revisit this

Reply:
Fixed